### ISSUE TITLE
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "". Strict MIME type checking is enforced for module scripts per HTML spec.

### ISSUE THREAD

**Issue by moshi-soft** [nuxt/framework#5303]
In nuxt 3 After running **nuxt generate** I am unable to run the project generated in .output folders.
showing error like:
**_Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "". Strict MIME type checking is enforced for module scripts per HTML spec._**

Actually I want to host the project on cpanel.
How can I solve the problem?

In short I actually want to host the my nuxt 3 project on cpanel

**Comment by IainZhuo** [comment]
使用Nginx部署，遇到相同问题

**Comment by manniL** [comment]
Did you use `target: static` in your `nuxt.config.ts` (or `.js`)?
Can you share a full reproduction?

Also, this seems like it should be a [discussion](https://github.com/nuxt/framework/discussions) instead ☺️

**Comment by moshi-soft** [comment]
@manniL I have tried using `target: static` still same problem

**Comment by alex-maereanu** [comment]
Same problem here .. any idea anyone?

**Comment by alex-maereanu** [comment]
Using NGINX on the deployment server

**Comment by alex-maereanu** [comment]
Ok guys: just add this to your MIME types in NGINX configuration:
`application/javascript  mjs;`
 and everything will work as intended. It looks that NGINX doesn't offer MIME types for mjs files OOTB

**Comment by moshi-soft** [comment]
Thanks @alex-maereanu 
in apache 
```
<IfModule mod_mime.c>
  AddType text/javascript js mjs
</IfModule>
```
inside .htaccess

**Comment by Silbad** [comment]
Hello, and if I can't change the server configuration, what can I do? Can we build in simple js files?

**Comment by Eliophot** [comment]
Hello @moshi-soft, Apache method don't work for me...

**Comment by moshi-soft** [comment]
@Silbad @Eliophot If you are using apache server then .htaccess is a file for product/application based configuration file.
Create a .htaccess file inside your application and paste 
```
<IfModule mod_mime.c>
  AddType text/javascript js mjs
</IfModule>
```
If doesn't work make sure your apache mod_rewrite is enabled

**Comment by Eliophot** [comment]
@moshi-soft thanks, yes I did that but it doesn't work (I put the htaccess in the root of the server)
```
<IfModule mod_rewrite.c>
    RewriteCond %{HTTP:X-Forwarded-Proto} !https
    RewriteCond %{HTTPS} off
    RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]

    RewriteCond %{HTTP_HOST} !^www\. [NC]
    RewriteRule (.*) https://www.%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
</IfModule>
<IfModule mod_mime.c>
  AddType text/javascript js mjs
</IfModule>
```

**Comment by Eliophot** [comment]
@danielroe
I had this in nuxt.config.ts
```
    hooks: {
        'vite:extendConfig'(config, { isClient }) {
            if (process.env.NODE_ENV !== 'development' && isClient) {
                // rc6
                config.build.rollupOptions.output.chunkFileNames = '[hash].js'
                // rc7
                config.build.rollupOptions.output.chunkFileNames = '_nuxt/[hash].js'
            }
        }
    },
```
thanks, js files are build ^^
But _nuxt folder is call two times : 
![image](https://user-images.githubusercontent.com/81296123/184141037-84958dc9-65d6-4af0-bf97-459c06a3f923.png)
Any ideas ?

**Comment by danielroe** [comment]
you are meant to pick a line based on whether you have rc6 or rc7 installed

**Comment by Eliophot** [comment]
(-_-) sorry
ok I'll try with the new rc7

**Comment by Eliophot** [comment]
Test rc6
![image](https://user-images.githubusercontent.com/81296123/184148200-51aa5e35-a3f3-4744-8519-70755135b437.png)

**Comment by danielroe** [comment]
Yes. You need to use RC7 with new vue-bundle-renderer to have a `.js` file extension.

**Comment by Eliophot** [comment]
Test rc7, only one more error, we're almost there ^^
![image](https://user-images.githubusercontent.com/81296123/184156720-60adbb82-f85e-4903-8960-2ca7fb52cdb2.png)

**Comment by Eliophot** [comment]
Ok, it works !! with : 
```js
    hooks: {
        'vite:extendConfig'(config, { isClient }) {
            if (process.env.NODE_ENV !== 'development' && isClient) {
                config.build.rollupOptions.output.chunkFileNames = '_nuxt/[hash].js'
                config.build.rollupOptions.output.entryFileNames = '_nuxt/entry.[hash].js'
            }
        }
    },
```
Thanks !!!!!!

**Comment by arthurducept** [comment]
> Ok, it works !! with :
> 
> ```js
>     hooks: {
>         'vite:extendConfig'(config, { isClient }) {
>             if (process.env.NODE_ENV !== 'development' && isClient) {
>                 config.build.rollupOptions.output.chunkFileNames = '_nuxt/[hash].js'
>                 config.build.rollupOptions.output.entryFileNames = '_nuxt/entry.[hash].js'
>             }
>         }
>     },
> ```
> 
> Thanks !!!!!!

I've got this error when I try your solution :/

![image](https://user-images.githubusercontent.com/48486020/184951766-e28398bb-d5d4-4ae9-8362-9f5eb5d6540b.png)

Could you share your nuxt.config.ts (or .js) ?

**Comment by danielroe** [comment]
@arthurducept Ensure you are on rc8, or you can omit the _nuxt prefix.

**Comment by arthurducept** [comment]
@danielroe I didn't have to, i just wrote the mime_types in the wrong order in my nginx configuration. It worked without changing the nuxt.config.ts after that. I edited my previous comment, hoping it will help someone else

**Comment by lboh-onv** [comment]
In my case, this problem appeared after upgrading the rc-version and deleting the `.nuxt/dist` directory fixed the issue

**Comment by v-romanyuk** [comment]
@danielroe I'm still having that issue in the generate mode on rc8.

**Comment by danielroe** [comment]
@gezichenshan https://github.com/nuxt/framework/issues/5303#issuecomment-1212031047.

**Comment by yuxufm** [comment]
in Apache server.

```bash
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream". Strict MIME type checking is enforced for module scripts per HTML spec.
```

You need to usa an .htaccess file or extend your Apache config file with the following htaccess directive in order for the server to output files with the mjs extension with the correct MIME type:

```bash
<IfModule mod_mime.c>
  AddType text/javascript js mjs
</IfModule>
```

Of course, mod_mime needs to be installed and enabled for this to work.

**Comment by wzc520pyfm** [comment]
I had a similar problem with nuxt3.0.0-rc.11, but it disappeared after I cleared the browser cache.
